src/legacy/core_plugins/metrics/server/lib/vis_data/get_annotations.js

```diff
@@ -44,7 +46,7 @@ export async function getAnnotations(req, panel, esQueryConfig, searchStrategy,
 
     return annotations
       .reduce((acc, annotation, index) => {
-        acc[annotation.id] = handleAnnotationResponse(responses[index], annotation);
+        acc[annotation.id] = handleAnnotationResponse(lastSeriesTimestamp)(responses[index], annotation);
```

Comment:
What about to move the **handleAnnotationResponse(lastSeriesTimestamp)** from the 'reduce' cycle
```
  const handleResponse = handleAnnotationResponse(lastSeriesTimestamp);
  return annotations
      .reduce((acc, annotation, index) => {
        acc[annotation.id] = handleResponse(responses[index], annotation);
```

Reply:
It makes sense. Done